src/commands/cronjob/update.tsx

```diff
@@ -0,0 +1,153 @@
+import { ExecRenderBaseCommand } from "../../lib/basecommands/ExecRenderBaseCommand.js";
+import { Args, Flags } from "@oclif/core";
+import { ReactNode } from "react";
+import {
+  makeProcessRenderer,
+  processFlags,
+} from "../../rendering/process/process_flags.js";
+import { Success } from "../../rendering/react/components/Success.js";
+import assertSuccess from "../../lib/apiutil/assert_success.js";
+import { MittwaldAPIV2 } from "@mittwald/api-client";
+
+type UpdateResult = void;
+
+export default class Update extends ExecRenderBaseCommand<
+  typeof Update,
+  UpdateResult
+> {
+  static description = "Update an existing cron job";
+  static args = {
+    "cronjob-id": Args.string({
+      description: "ID of the cron job to be updated.",
+      required: true,
+    }),
+  };
+  static flags = {
+    description: Flags.string({
+      description: "Set cron job description",
+    }),
+    interval: Flags.string({
+      description: "Set cron job execution interval",
+    }),
+    disable: Flags.boolean({
+      description: "Disable cron job automated execution",
+      exclusive: ["enable"],
+    }),
+    enable: Flags.boolean({
+      description: "Enable cron job automated execution",
+      exclusive: ["disable"],
+    }),
+    email: Flags.string({
+      description: "Set target email to send error messages to",
+    }),
+    timeout: Flags.integer({
+      description: "Set timeout in seconds after wich the process is killed",
+    }),
+    url: Flags.string({
+      description: "Set url to use on cron job execution",
+    }),
+    command: Flags.string({
+      description: "Set file and parameters to execute on cron job execution",
+    }),
+    interpreter: Flags.string({
+      description: "Set interpreter to use for execution",
+      options: ["bash", "php"],
+    }),
+    ...processFlags,
+  };
+
+  protected async exec(): Promise<void> {
+    type destinationType =
+      | MittwaldAPIV2.Components.Schemas.CronjobCronjobCommand
+      | MittwaldAPIV2.Components.Schemas.CronjobCronjobUrl;
+
+    type cronjobUpdatePayload = {
+      active: boolean;
+      description: string;
+      destination: destinationType;
+      email: string;
+      timeout: number;
+      interval: string;
+    };
```

Comment:
For the entire PR: Is there any way to extract these types from the API client package, so we don't need to duplicate them here?

Reply:
this has been fixed for multiple PRs